File: 1961. Check If String Is a Prefix of Array/solution.js
/**
 * @param {string} s
 * @param {string[]} words
 * @return {boolean}
 */
 var isPrefixString = function(s, words) {
  let currentStructure = "";
  for (let word of words){
    currentStructure += word;
    if (currentStructure === s){
      return true;
    }
    if (currentStructure.length > s.length){
      break;
    }
  }

  return false;
};

File: 1961. Check If String Is a Prefix of Array/solution.java
class Solution {
  public boolean isPrefixString(String s, String[] words) {
    String currentStructure = "";
    for (int i = 0; i < words.length; i++){
      currentStructure += words[i];
      if (currentStructure.equals(s)){
        return true;
      }
      if (currentStructure.length() > s.length()){
        break;
      }
    }
    return false;
  }
}